common_types/src/datum.rs

```diff
@@ -347,6 +382,14 @@ pub enum Datum {
     Int16(i16),
     Int8(i8),
     Boolean(bool),
+    /// The Date representing the elapsed time since UNIX epoch in days(32 bits)
+    /// It is mapped to [`arrow::datatypes::DataType::Date32`].
+    /// The supported date range is '-9999-01-01' to '9999-12-31'.
+    Date(i32),
+    /// The Time representing the elapsed time since midnight in nanoseconds(64
+    /// bits) It is mapped to [`arrow::datatypes::DataType::Time64`].
```

Comment:
```suggestion
    /// Time represents the elapsed nanoseconds since midnight. 
    ///
    /// It is mapped to [`arrow::datatypes::DataType::Time64`].
```